update architecture diagram (#242)
digital-asset/docs.daml.com: docs/2.6.3/docs/app-dev/app-arch.rst

.. image:: ./recommended_architecture.svg
   :alt: Diagram of the recommended application architecture, explained in depth immediately below.

> The above image shows the recommended Daml solution architecture. Here there are four types of building blocks that go into our application: user code, generated code from Daml, Daml components,  and external components.

Reviewer: ```suggestion
The above image shows the recommended Daml solution architecture. Here there are four types of building blocks that go into our application: user code, generated code from Daml, Daml components, and external components.
```

> In the recommended architecture, the Daml model determines the DAR files that underpin both the frontend and backend. The frontend includes user code such as a React Web Frontend, Daml React libraries or other integration libraries, and generated code from the DAR files (TypeScript).   It does not change things if a client service, instead of a GUI frontend, accesses the Daml application backend.

Reviewer: ```suggestion
In the recommended architecture, the Daml model determines the DAR files that underpin both the frontend and backend. The frontend includes user code such as a React Web Frontend, Daml React libraries or other integration libraries, and generated code from the DAR files (TypeScript). A client service can access the Daml application backend instead of a GUI frontend with no change to the rest of the architecture.
```
